client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -159,29 +181,137 @@ public boolean ping(int timeout) {
+            }
+        }
+        this.getterMethods.put(clazz, getterMethods);//Store the getter methods for later use
+
+        for (ClickHouseColumn column : schema.getColumns()) {
+            String columnName = column.getColumnName().toLowerCase().replace("_", "");
+            serializers.add((obj, stream) -> {
+                if (!getterMethods.containsKey(columnName)) {
+                    LOG.warn("No getter method found for column: {}", columnName);
+                    return;
+                }
+                Method getterMethod = this.getterMethods.get(clazz).get(columnName);
+                Object value = getterMethod.invoke(obj);
+                boolean hasDefaults = this.hasDefaults.get(clazz);
+
+                //Handle null values
+                if (value == null) {
+                    if (hasDefaults && !column.hasDefault()) {//Send this only if there is no default
+                        BinaryStreamUtils.writeNonNull(stream);
+                    }
+                    BinaryStreamUtils.writeNull(stream);//We send this regardless of default or nullable
+                    return;
+                }
+
+                //Handle default
+                if (hasDefaults) {
+                    BinaryStreamUtils.writeNonNull(stream);//Write 0
+                }
+
+                //Handle nullable
+                if (column.isNullable()) {
+                    BinaryStreamUtils.writeNonNull(stream);//Write 0
+                }
+
+                //Handle the different types
+                SerializerUtils.serializeData(stream, value, column);
+            });
+        }
+        this.serializers.put(clazz, serializers);
+        this.hasDefaults.put(clazz, schema.hasDefaults());
     }
 
     /**
      * Insert data into ClickHouse using a POJO
      */
-    public Future<InsertResponse> insert(String tableName,
-                                     List<Object> data,
-                                     InsertSettings settings,
-                                     List<ClickHouseColumn> columns) throws ClickHouseException, SocketException {
+    public InsertResponse insert(String tableName,
+                                         List<Object> data,
+                                         InsertSettings settings) throws IOException, InvocationTargetException, IllegalAccessException, ExecutionException, InterruptedException {
+        if (data == null || data.isEmpty()) {
+            throw new IllegalArgumentException("Data cannot be empty");
+        }
+        long s1 = System.currentTimeMillis();
+
+        //Add format to the settings
+        if (settings == null) {
+            settings = new InsertSettings();
+        }
+
+        boolean hasDefaults = this.hasDefaults.get(data.get(0).getClass());
+        if (hasDefaults) {
+            settings.setFormat(ClickHouseFormat.RowBinaryWithDefaults);
+        } else {
+            settings.setFormat(ClickHouseFormat.RowBinary);
```

Comment:
I think it should be RowBinaryWithNames because:
- RowBinary requires listing all columns and 
- We do not know if a POJO is fully mapped to a table